### DISCUSSION TITLE
pub/sub gets unmatch notification and participant discovery (left) notification very slow

### DISCUSSION THREAD

**Discussion by rookybird** [eProsima/Fast-DDS#3269] [General]
I am using Fast DDS 2.9.1.
After the initial pub/sub match, when the publisher is removed, the subscriber get the unmatched notification after like 20 seconds. Also, the participant left notification is also lagging 20 seconds. These two notifications show up at the same time.

I am wondering if I can make it quicker or instantaneous? Can anyone point me to the right direction where I should start digging?

**Comment by MiguelCompany** [comment]
This most probably means you are not removing the publisher in an ordered manner (i.e. you are not calling delete_datawriter / delete_participant). Most probably, you are just killing the process.

If that is the case, the only way for the other process to know the participant is gone, is with the participant's lease duration, which is by default 20 seconds.

**Reply by rookybird** [reply]
> This most probably means you are not removing the publisher in an ordered manner (i.e. you are not calling delete_datawriter / delete_participant). Most probably, you are just killing the process.
> 
> If that is the case, the only way for the other process to know the participant is gone, is with the participant's lease duration, which is by default 20 seconds.

Thank you! That makes sense
